src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerFileReference.cs

```diff
@@ -617,11 +620,33 @@ private ImmutableArray<TExtension> GetAnalyzersForTypeNames(Assembly analyzerAss
                     TExtension? analyzer = typeInstance as TExtension ?? _coerceFunction?.Invoke(typeInstance);
                     if (analyzer != null)
                     {
-                        analyzers.Add(analyzer);
+                        analyzers.Add((typeName, analyzer));
                     }
                 }
 
-                return analyzers.ToImmutable();
+                analyzers.Sort(static (x, y) => string.Compare(x.typeName, y.typeName, StringComparison.OrdinalIgnoreCase));
```

Comment:
Since we are sorting here, I changed the type of `analyzerTypeNames` to `ImmutableHashSet` from `ImmutableSortedSet`